File: Bronze/2018/December/01_mixing_milk/MixingMilk2.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.PrintWriter;
import java.util.StringTokenizer;

/**
 * @author Red
 * @version 1.0
 */
public class MixingMilk2 {
    static int[] bucketCapacity = new int[3];
    static int[] bucketCurMilk = new int[3];

    public static void main(String[] args) throws Exception {
        BufferedReader r = new BufferedReader(new FileReader("mixmilk.in"));
        PrintWriter pw = new PrintWriter("mixmilk.out");
        for (int i = 0; i < 3; i++) {
            StringTokenizer st = new StringTokenizer(r.readLine());
            bucketCapacity[i] = Integer.parseInt(st.nextToken());
            bucketCurMilk[i] = Integer.parseInt(st.nextToken());
        }
        for(int i = 0; i < 100; i++) {
            int from = i % 3;
            int to = (i + 1) % 3;
            int mount = Math.min(bucketCurMilk[from], bucketCapacity[to] - bucketCurMilk[to]);
            bucketCurMilk[from] -= mount;
            bucketCurMilk[to] += mount;
        }
        for (int i = 0; i < 3; i++) {
            pw.println(bucketCurMilk[i]);
        }
        pw.close();
    }
}

File: Bronze/2018/December/01_mixing_milk/mixing_milk2.cpp
#include <iostream>
#include <fstream>

using namespace std;

int bucket_capacity[3];
int bucket_cur_milk[3];

int main() {
    ifstream fin("mixmilk.in");
    ofstream fout("mixmilk.out");

    for (int i = 0; i < 3; i++) {
        fin >> bucket_capacity[i] >> bucket_cur_milk[i];
    }
    // 倒 100  次牛奶
    for (int i = 0; i < 100; i++) {
        int from = i % 3;
        int to = (i + 1) % 3;
        int pour_milk = min(bucket_cur_milk[from], bucket_capacity[to] - bucket_cur_milk[to]);
        bucket_cur_milk[from] -= pour_milk;
        bucket_cur_milk[to] += pour_milk;
    }

    for (int i = 0; i < 3; i++) {
        fout << bucket_cur_milk[i] << endl;
    }

    fin.close();
    fout.close();

    return 0;
}
